### ISSUE TITLE
Retain Rig Details After Restart

### ISSUE THREAD

**Issue by LanceMcCarthy** [nicehash/NiceHashQuickMiner#140]
When I need to manually stop and restart nhqm, it shows up as a new rig. This leaves a bunch of abandoned rigs in my monitoring and stats. I've had to delete 4 abandoned rigs now and it's messing with my stats.

Is there a way you can please persist the client ID for the rig so that it is not recreated as a new rig with the same name?

![image](https://user-images.githubusercontent.com/3520532/110129964-841d2100-7d96-11eb-8f86-59a056b231f9.png)

**Comment by nicehashdev** [comment]
Are you deleting `nhqm.conf` file? Your unique ID is saved there which identifies your rig and links it with your statistics online. If unique ID gets removed/deleted, changed somehow, then your rig will appear as new one. You need to make sure that your unique ID is not changed and you will not have this problem anymore.

**Comment by LanceMcCarthy** [comment]
I'm not intentionally deleting that file, it just seems to happen when I manually exit nhqm and restart it by using the shortcut the installer placed on my desktop. 

Should I be using a different method to start the app (after it has been installed and configured)?

**Comment by nicehashdev** [comment]
Did you install NHQM into a location where then, it doesn't have access to write to? Eg, you prevent NHQM to be run as Administrator? Then this would explain this behavior.

**Comment by LanceMcCarthy** [comment]
I'll double check, I've been running the different versions for a while now and the app has evolved significantly since then, so I lost track of what steps I've taken.

I used to run it as a standalone exe, right from the extracted ZIP folder in user/downloads (because starting the process using NT System or Network Service was a problem). After the latest update I just used the installer as is and start it from the shortcut.

What can't remember is if I get a UAC prompt after launching from the desktop shortcut. I think it just starts right away.

**Comment by LanceMcCarthy** [comment]
I can reproduce the problem 100% of the time. Let me take you through it

Here are the steps

1. Prepare a system with a running miner (confirmed working and has a rig name)
2. Restart the PC
2. After reboot, observe the tray has NHQM already running, but it is not started
3. Right-click NHQM icon and select Start

### Problems

At this point, it is stuck. It states it is running

![image](https://user-images.githubusercontent.com/3520532/110189265-8316de80-7dec-11eb-8cce-da51fa0d6e48.png)


Looking at the output, there's a problem, no jobs are accepted:

![image](https://user-images.githubusercontent.com/3520532/110189326-acd00580-7dec-11eb-812a-ee0b64623f56.png)

If you Right-click on NHQM and click  "Stop", then click "Start" 

the problem still exists, there must be an issue with the user running the process (I forgot to check in Task Manager who owns the process).

![image](https://user-images.githubusercontent.com/3520532/110189387-eacd2980-7dec-11eb-8302-68c8d5f1a0d3.png)


### Workaround and Abandoned Rig

In order to fix this, I need to kill NHQM via the tray

![image](https://user-images.githubusercontent.com/3520532/110189416-00daea00-7ded-11eb-8ed9-2b4031d3aaa4.png)

I relaunch NHQM by double clicking the desktop shortcut

![image](https://user-images.githubusercontent.com/3520532/110189449-1b14c800-7ded-11eb-9ca6-17d57d381d7e.png)


I approve the UAC prompt and the NHQM starts working properly

![image](https://user-images.githubusercontent.com/3520532/110189483-35e73c80-7ded-11eb-816b-d31422118137.png)



I look in the Rig Details page in the website and the last Rig is now abandoned

![image](https://user-images.githubusercontent.com/3520532/110189580-8363a980-7ded-11eb-9f4e-aea70c14a5db.png)



### Suspicion

Why is NHQM already running after reboot? I suspect a task started it that doesn't have elevated rights, which might explain the network communication issue trying to get jobs.

That doesn't solve the problem where the rig gets a new client ID though. IF it were an elevated process issue, then the original ID from when it was working before the reboot should still be the same after manually launching it.

**Comment by nicehashdev** [comment]
Make sure you don't use service method and remove all services that have name NiceHash QuickMiner.

Also check your Task Scheduler. Do you have NiceHash QuickMiner there?

Do you have some special version of Windows with manual modified access for users regarding UAC? If yes, than this could explain this behavior. NiceHash QuickMiner needs elevated privileges to run due to various OC commands that can be executed only when being elevated.

**Comment by LanceMcCarthy** [comment]
Your idea pointed me to the cause:

![image](https://user-images.githubusercontent.com/3520532/110190290-8e6c0900-7df0-11eb-94ad-bf23f4c4e77b.png)

IIRC the new autostart feature doesn't use a Windows Service anymore, right? So is this a task I can manually edit to let it run as User and admin rights?

If not, I will switch it over to a custom Task Scheduler task that has user account and elevated permissions.

**Comment by nicehashdev** [comment]
New versions do check for service method and if it is present, try to remove it. But if you run NHQM without elevated privileges, then this action cannot succeed. That is why I suggest you to check services just in case (Windows taskbar search box, write: "services" and hit enter then try to find NiceHash QuickMiner in the list - if there is such service, remove it!)

New version use only Task Scheduler autostart which has turned out to be pretty reliable. It also starts with elevated privileges without any issues. But for this task to be added, NHQM needs to have elevated privileges.

**Comment by LanceMcCarthy** [comment]
Agreed, I personally use Task Scheduler when I need reliable automated operation with a specific user and specific permissions. It's perfect for bat scripts.

Okay, to close out this Issue:

I checked the Services list and NHQM was not there, so that's cleared ☑
I unchecked the Start with Windows setting and it deleted the existing task ☑
I re-checked the "start with windows" setting and it added the Task. ☑

Finally, I opened Task Scheduler, found the task and verified the user and elevated settings ✅🏁

![image](https://user-images.githubusercontent.com/3520532/110192216-a6e12100-7dfa-11eb-8043-52a44a328bbd.png)
